File: src/main/Java/boj_step9/boj9506.java
import java.io.*;
import java.util.LinkedList;
import java.util.Queue;

public class boj9506 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        int n = 0;
        while ((n = Integer.parseInt(reader.readLine())) != -1) {
            Queue<Integer> factors = new LinkedList<>();
            for (int i = 1; i <= n / 2; i++) if (n % i == 0) factors.add(i);
            if (n == factors.stream().mapToInt(Integer::intValue).sum()) {
                writer.append(n + " = " + factors.poll());
                while (!factors.isEmpty()) writer.append(" + " + factors.poll());
            } else writer.append(n + " is NOT perfect.");
            writer.newLine();
        }
        writer.close();
    }
}

File: src/main/Kotlin/boj_step9/boj9506.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.lang.System

fun main() {
    val r = BufferedReader(InputStreamReader(System.`in`))
    val w = BufferedWriter(OutputStreamWriter(System.out))
    while (true) {
        val n = r.readLine().toInt()
        if (n == -1) {
            w.close()
            return
        }
        val factors = mutableListOf<Int>()
        for (i in 1..n / 2) if (n % i == 0) factors.add(i)
        if (factors.sum() == n) {
            w.write("$n = ")
            factors.forEachIndexed { index, it ->
                if (index == factors.size - 1) w.write("$it")
                else w.write("$it + ")
            }
        } else w.write("$n is NOT perfect.")
        w.write("\n")
    }
}